ambry-store/src/test/java/com.github.ambry.store/CompactionLogTest.java

```diff
@@ -0,0 +1,296 @@
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ */
+package com.github.ambry.store;
+
+import com.github.ambry.utils.MockTime;
+import com.github.ambry.utils.TestUtils;
+import com.github.ambry.utils.Time;
+import com.github.ambry.utils.Utils;
+import com.github.ambry.utils.UtilsTest;
+import java.io.File;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.UUID;
+import org.junit.After;
+import org.junit.Test;
+
+import static org.junit.Assert.*;
+
+
+/**
+ * Tests for {@link CompactionLog}.
+ */
+public class CompactionLogTest {
+  private static final StoreKeyFactory STORE_KEY_FACTORY;
+
+  static {
+    try {
+      STORE_KEY_FACTORY = new MockIdFactory();
+    } catch (Exception e) {
+      throw new IllegalStateException(e);
+    }
+  }
+
+  // Variables that represent the folder where the data resides
+  private final File tempDir;
+  private final String tempDirStr;
+  // the time instance that will be used in the index
+  private final Time time = new MockTime();
+
+  /**
+   * Creates a temporary directory for the compaction log file.
+   * @throws IOException
+   */
+  public CompactionLogTest() throws IOException {
+    tempDir = StoreTestUtils.createTempDirectory("storeDir-" + UtilsTest.getRandomString(10));
+    tempDirStr = tempDir.getAbsolutePath();
+  }
+
+  /**
+   * Deletes the temporary directory.
+   * @throws InterruptedException
+   * @throws IOException
+   */
+  @After
+  public void cleanup() throws InterruptedException, IOException, StoreException {
+    assertTrue(tempDir.getAbsolutePath() + " could not be deleted", StoreTestUtils.cleanDirectory(tempDir, true));
+  }
+
+  /**
+   * Tests the use of {@link CompactionLog} when it is used without closing b/w operations and compaction cycles.
+   * @throws IOException
+   */
+  @Test
+  public void iterationWithoutReloadTest() throws IOException {
+    String storeName = "store";
+    List<CompactionDetails> detailsList = getCompactionDetails(5);
+    assertFalse("Compaction should not be in progress", CompactionLog.isCompactionInProgress(tempDirStr, storeName));
+    CompactionLog log = new CompactionLog(tempDirStr, storeName, time, detailsList);
+    assertTrue("Compaction should should be in progress", CompactionLog.isCompactionInProgress(tempDirStr, storeName));
+    for (CompactionDetails details : detailsList) {
+      verifyEquality(details, log.getCompactionDetails());
+      assertEquals("Should be in the PREPARE phase", CompactionLog.Phase.PREPARE, log.getCompactionPhase());
+      log.markCopyStart();
+      assertEquals("Should be in the COPY phase", CompactionLog.Phase.COPY, log.getCompactionPhase());
+      StoreFindToken safeToken =
```

Comment:
this might be too much to validate(may be). But do you think validating if storeFindTonken lies within the range of logSegmentsToCompact for that cycle is required?